### PR TITLE
Add sas_token lifetime option to iothub sdk

### PR DESCRIPTION
**Pull request by jebrando**

### CODE REVIEW

File: iothub_client/src/iothubtransport_mqtt_common.c [hunk 1]
```diff
@@ -2583,6 +2585,12 @@ IOTHUB_CLIENT_RESULT IoTHubTransport_MQTT_Common_SetOption(TRANSPORT_LL_HANDLE h
             mqtt_client_set_trace(transport_data->mqttClient, transport_data->log_trace, transport_data->raw_trace);
             result = IOTHUB_CLIENT_OK;
         }
+        else if (strcmp(OPTION_SAS_TOKEN_LIFETIME, option) == 0)
+        {
+            size_t* sas_lifetime = (size_t*)value;
```
Comment (anthonyvercolano) [inline iothub_client/src/iothubtransport_mqtt_common.c]: >size_t* sas_lifetime =  [](start = 12, length = 23)

Why doing this?

File: iothub_client/tests/iothubtransport_mqtt_common_ut/iothubtransport_mqtt_common_ut.c [hunk 2]
```diff
@@ -2169,6 +2169,49 @@ TEST_FUNCTION(IoTHubTransport_MQTT_Common_SetOption_value_NULL_fail)
     IoTHubTransport_MQTT_Common_Destroy(handle);
 }
 
+TEST_FUNCTION(IoTHubTransport_MQTT_Common_SetOption_SAS_TOKEN_LIFETIME_NULL_fail)
+{
```
Comment (tameraw) [inline iothub_client/tests/iothubtransport_mqtt_common_ut/iothubtransport_mqtt_common_ut.c]: Missing requirement updates.

File: iothub_client/tests/iothubtransport_mqtt_common_ut/iothubtransport_mqtt_common_ut.c [hunk 3]
```diff
@@ -2169,6 +2169,49 @@ TEST_FUNCTION(IoTHubTransport_MQTT_Common_SetOption_value_NULL_fail)
     IoTHubTransport_MQTT_Common_Destroy(handle);
 }
 
+TEST_FUNCTION(IoTHubTransport_MQTT_Common_SetOption_SAS_TOKEN_LIFETIME_NULL_fail)
```
Comment (tameraw) [inline iothub_client/tests/iothubtransport_mqtt_common_ut/iothubtransport_mqtt_common_ut.c]: >IoTHubTransport_MQTT_Common_SetOption_ [](start = 14, length = 38)

Can we also please update the setoption comments in _LL with the option key/value?

File: iothub_client/src/iothubtransport_mqtt_common.c (hunk 1 above)
Comment (jebrando) [reply iothub_client/src/iothubtransport_mqtt_common.c]: Github issue

File: iothub_client/tests/iothubtransport_mqtt_common_ut/iothubtransport_mqtt_common_ut.c (hunk 3 above)
Comment (jebrando) [reply iothub_client/tests/iothubtransport_mqtt_common_ut/iothubtransport_mqtt_common_ut.c]: Sounds good.

File: iothub_client/tests/iothubtransport_mqtt_common_ut/iothubtransport_mqtt_common_ut.c (hunk 2 above)
Comment (jebrando) [reply iothub_client/tests/iothubtransport_mqtt_common_ut/iothubtransport_mqtt_common_ut.c]: Updated.

File: iothub_client/src/iothubtransport_mqtt_common.c (hunk 1 above)
Comment (jebrando) [reply iothub_client/src/iothubtransport_mqtt_common.c]: Just to be specific.

---
In reply to: [140592649](https://github.com/Azure/azure-iot-sdk-c/pull/240#discussion_r140592649) [](ancestors = 140592649)

**Pull request merged**